src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs

```diff
@@ -779,45 +777,24 @@ internal void UpdateProxyPortsPosition()
 
             if (parent != null && !parent.IsExpanded) return;
 
-            IEnumerable<NodeModel> nestedNodes = null;
-            if (annotationModel.HasNestedGroups)
+            for (int i = 0; i < inPorts.Count(); i++)
             {
-                nestedNodes = Nodes
-                    .OfType<AnnotationModel>()
-                    .SelectMany(x => x.Nodes.OfType<NodeModel>())
-                    .Concat(Nodes.OfType<NodeModel>());
-
-                // If any of the nested groups are collapsed
-                // we need to update the posistion of the 
-                // proxy ports on it.
-                foreach (var nestedGroup in ViewModelBases.OfType<AnnotationViewModel>().Where(x => !x.IsExpanded))
+                var model = inPorts[i].PortModel;
+                if (model.IsProxyPort)
                 {
-                    nestedGroup.SetGroupInputPorts();
-                    nestedGroup.SetGroupOutPorts();
-                    nestedGroup.UpdateProxyPortsPosition();
+                    // calculate new position for the proxy inports.
+                    model.Center = CalculatePortPosition(model, i);
                 }
             }
 
-            var groupInports = GetGroupInPorts(IsExpanded ? null : nestedNodes);
-
-            for (int i = 0; i < groupInports.Count(); i++)
+            for (int i = 0; i < outPorts.Count(); i++)
```

Comment:
Can we just use foreach?

Reply:
The index is needed in a function call a bit lower